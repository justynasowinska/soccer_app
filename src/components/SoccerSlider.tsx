import * as React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import Slider from '@react-native-community/slider';

import { formatMsToString } from '../utils/dataUtils';

interface PropsType {
    currentTime: number;
    totalTime: number;
    value: number;
    onValueChange: (value: number) => void;
}

export const SoccerSlider = (props: PropsType) => {
    const { currentTime, totalTime, onValueChange, value } = props;

    return (
        <View style={styles.container}>
            <Text style={[styles.timeText, { left: 0 }]}>{formatMsToString(currentTime)}</Text>
            <Slider
                maximumValue={totalTime}
                minimumValue={0}
                onValueChange={onValueChange}
                value={value}
            />
            <Text style={[styles.timeText, { right: 0 }]}>{formatMsToString(totalTime)}</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        position: 'relative',
        width: '100%',
    },
    timeText: {
        position: 'absolute',
        bottom: -10,
        fontSize: 13,
    }
});