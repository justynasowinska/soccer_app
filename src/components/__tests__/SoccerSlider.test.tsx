import * as React from 'react';
import { shallow } from 'enzyme';

import { SoccerSlider } from '../SoccerSlider';

describe('SoccerSlider', () => {
    it('matches to the snapshot', () => {
        const component = shallow(
            <SoccerSlider
                currentTime={1434}
                totalTime={23454}
                value={1222}
                onValueChange={jest.fn()}
            />
        );

        expect(component).toMatchSnapshot();
    });
});