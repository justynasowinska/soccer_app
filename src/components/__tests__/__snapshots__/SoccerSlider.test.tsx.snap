// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoccerSlider matches to the snapshot 1`] = `
<View
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "bottom": -10,
          "color": "#F6F6F6",
          "fontSize": 13,
          "fontWeight": "bold",
          "position": "absolute",
        },
        Object {
          "left": 0,
        },
      ]
    }
  >
    00:00:01
  </Text>
  <ForwardRef(Themed.Slider)
    maximumTrackTintColor="rgba(255, 255, 255, 0.4)"
    maximumValue={23454}
    minimumTrackTintColor="#00B4CC"
    minimumValue={0}
    onValueChange={[MockFunction]}
    thumbTintColor="#00B4CC"
    thumbTouchSize={
      Object {
        "height": 70,
        "width": 70,
      }
    }
    value={1222}
  />
  <Text
    style={
      Array [
        Object {
          "bottom": -10,
          "color": "#F6F6F6",
          "fontSize": 13,
          "fontWeight": "bold",
          "position": "absolute",
        },
        Object {
          "right": 0,
        },
      ]
    }
  >
    00:00:23
  </Text>
</View>
`;
